lib/inets/test/httpc_SUITE.erl

```diff
@@ -252,6 +258,24 @@ end_per_testcase(pipeline, _Config) ->
     inets:stop(httpc, pipeline);
 end_per_testcase(persistent_connection, _Config) ->
     inets:stop(httpc, persistent);
+end_per_testcase(Case, Config)
+  when Case == server_closing_connection_on_first_response;
+       Case == server_closing_connection_on_second_response ->
+    %% Test case uses at most one session.  Ensure no leftover
+    %% sessions left behind.
+    {_, Status} = proplists:lookup(tc_status, Config),
+    ShallCleanup = case Status of
+                       ok -> true;
+                       {failed, _} -> true;
+                       {skipped, _} -> false
+                   end,
+    if ShallCleanup ->
+            httpc:request(url(group_name(Config), "/just_close.html", Config)),
+            ok;
+       true ->
```

Comment:
This clause cannot match as the previous clause `if ShallCleanup ->` will always match.

Reply:
> This clause cannot match as the previous clause if ShallCleanup -> will always match.

This looks like a Dialyzer limitation?

> Possible fix:
> 
> if ShallCleanup =:= true ->

Worked around as you suggested in new commit.